// @mermaid-js/mermaid-zenuml/dist/chunks/mermaid-zenuml.esm/zoom-out.b73582c2-2D3Z2MQU.mjs
import {
  R,
  S,
  q
} from "./chunk-EQXVLZUD.mjs";
import {
  __name
} from "./chunk-FRNSN3NF.mjs";

// ../../node_modules/.pnpm/@zenuml+core@3.17.2_ts-node@10.9.2_typescript@5.3.3/node_modules/@zenuml/core/dist/zoom-out.b73582c2.js
var e = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
};
var r = /* @__PURE__ */ S("path", {
  fill: "currentColor",
  d: "M637 443H325c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h312c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
}, null, -1);
var l = /* @__PURE__ */ S("path", {
  fill: "currentColor",
  d: "M921 867 775 721c122.1-148.9 113.6-369.5-26-509-148-148.1-388.4-148.1-537 0-148.1 148.6-148.1 389 0 537 139.5 139.6 360.1 148.1 509 26l146 146c3.2 2.8 8.3 2.8 11 0l43-43c2.8-2.7 2.8-7.8 0-11zM696 696c-118.8 118.7-311.2 118.7-430 0-118.7-118.8-118.7-311.2 0-430 118.8-118.7 311.2-118.7 430 0 118.7 118.8 118.7 311.2 0 430z"
}, null, -1);
var n = [
  r,
  l
];
function s(a, d) {
  return R(), q("svg", e, n);
}
__name(s, "s");
var h = { render: s };
export {
  h as default,
  s as render
};
